src/set_theory/surreal/basic.lean

```diff
@@ -110,36 +110,31 @@ theorem lf_iff_lt {x y : pgame} (ox : numeric x) (oy : numeric y) : x ⧏ y ↔
 theorem le_iff_forall_lt {x y : pgame} (ox : x.numeric) (oy : y.numeric) :
   x ≤ y ↔ (∀ i, x.move_left i < y) ∧ ∀ j, x < y.move_right j :=
 begin
-  rw le_iff_forall_lf,
-  refine and_congr _ _;
-    refine forall_congr (λ i, (lf_iff_lt _ _));
-    apply_rules [numeric.move_left, numeric.move_right]
+  refine le_iff_forall_lf.trans (and_congr _ _);
+  refine forall_congr (λ i, lf_iff_lt _ _);
+  apply_rules [numeric.move_left, numeric.move_right]
 end
 
-theorem le_of_forall_lt {x y : pgame} (ox : x.numeric) (oy : y.numeric) :
-  ((∀ i, x.move_left i < y) ∧ ∀ j, x < y.move_right j) → x ≤ y :=
-(le_iff_forall_lt ox oy).2
-
 /-- Definition of `x < y` on numeric pre-games, in terms of `≤` -/
-theorem lt_iff_forall_le {x y : pgame} (ox : x.numeric) (oy : y.numeric) :
+theorem lt_iff_exists_le {x y : pgame} (ox : x.numeric) (oy : y.numeric) :
   x < y ↔ (∃ i, x ≤ y.move_left i) ∨ ∃ j, x.move_right j ≤ y :=
-by rw [←lf_iff_lt ox oy, lf_iff_forall_le]
+by rw [←lf_iff_lt ox oy, lf_iff_exists_le]
 
-theorem lt_of_forall_le {x y : pgame} (ox : x.numeric) (oy : y.numeric) :
+theorem lt_of_exists_le {x y : pgame} (ox : x.numeric) (oy : y.numeric) :
   ((∃ i, x ≤ y.move_left i) ∨ ∃ j, x.move_right j ≤ y) → x < y :=
-(lt_iff_forall_le ox oy).2
+(lt_iff_exists_le ox oy).2
 
 /-- The definition of `x < y` on numeric pre-games, in terms of `<` two moves later. -/
 theorem lt_def {x y : pgame} (ox : x.numeric) (oy : y.numeric) : x < y ↔
   (∃ i, (∀ i', x.move_left i' < y.move_left i)  ∧ ∀ j, x < (y.move_left i).move_right j) ∨
    ∃ j, (∀ i, (x.move_right j).move_left i < y) ∧ ∀ j', x.move_right j < y.move_right j' :=
 begin
   rw [←lf_iff_lt ox oy, lf_def],
-  refine or_congr _ _;
-    refine exists_congr (λ x_1, _);
-    refine and_congr _ _;
-    refine (forall_congr $ λ i, lf_iff_lt _ _);
-    apply_rules [numeric.move_left, numeric.move_right]
+  convert iff.rfl;
+  funext; congr; ext;
+  apply forall_congr (λ j, _);
+  rw lf_iff_lt;
+  apply_rules [numeric.move_left, numeric.move_right]
```

Comment:
Why did this change? Does the old proof not still work?

Note that the two-space inswnr is a good way to indicate "something weird is happening here with `;`s"

Reply:
Or phrased another way; this feels ungolfed to me, and certainly more fragile as both `convert` and `congr` aren't super predictable.

Reply:
The golfing rationale was that I could use `congr` instead of applying various `congr` lemmas (`exists_congr`, `and_congr`, `forall_congr`) in succession. It ended up not being as good of a golf as I expected, since `congr` doesn't seem to look inside lambdas, which means that neither the initial `funext` nor `forall_congr` are dealt with.

Still, I think this is at least a minor golf?

Reply:
I prefer the original version. This many `;`s is slightly frightening, and combined with scary tactics, it feels fragile for little gain.

Reply:
I've reverted this change.